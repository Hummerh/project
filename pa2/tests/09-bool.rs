fn main() {
      let a : bool = true;
      let b : bool = false;

      true || a || b && (a == b) || !a && !b;
      true == false || true != false;
}
