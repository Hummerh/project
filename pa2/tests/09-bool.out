(crate
    (items
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-id
                        (id
                            (a)))
                    (type-bool)
                    (true))
                (let
                    (pat-id
                        (id
                            (b)))
                    (type-bool)
                    (false))
                (or
                    (or
                        (or
                            (true)
                            (id
                                (a)))
                        (and
                            (id
                                (b))
                            (eq
                                (id
                                    (a))
                                (id
                                    (b)))))
                    (and
                        (not
                            (id
                                (a)))
                        (not
                            (id
                                (b)))))
                (or
                    (eq
                        (true)
                        (false))
                    (neq
                        (true)
                        (false)))))))